src/resources/2015/Dinsey.ts

```diff
@@ -0,0 +1,215 @@
+  ],
+  [
+    "questEStGiveMeFuel",
+    {
+      item: $item`toxic globule`,
+      quantity: 20,
+      trackingPref: "",
+      completeValue: 0,
+    },
+  ],
+  [
+    "questEStNastyBears",
+    {
+      item: $item`none`,
+      quantity: 0,
+      trackingPref: "dinseyNastyBearsDefeated",
+      completeValue: 8,
+    },
+  ],
+  [
+    "questEStWorkWithFood",
+    {
+      item: $item`complimentary Dinsey refreshments`,
+      quantity: 1,
+      trackingPref: "dinseyTouristsFed",
+      completeValue: 30,
+    },
+  ],
+]);
+
+export const keyCardsLocations = new Map<Item, Location>([
+  [$item`keycard α`, $location`Barf Mountain`],
+  [$item`keycard β`, $location`Pirates of the Garbage Barges`],
+  [$item`keycard γ`, $location`The Toxic Teacups`],
+  [
+    $item`keycard δ`,
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`,
+  ],
+]);
+
+export function acceptedQuest(): string | null {
+  const result = Array.from(questPreferenceKioskText.values()).find(
+    (questPref) => getString(questPref) !== "unstarted"
+  );
+  if (result) {
+    return result;
+  } else {
+    return null;
+  }
+}
+
+export function questFinished() {
+  const quest = acceptedQuest();
+  return quest && getString(quest) === "finished";
+}
+
+export function availableQuests(): string[] {
+  const kioskText = visitUrl(kioskUrl);
+  return Array.from(questPreferenceKioskText)
+    .filter((questPair) => kioskText.includes(questPair[1]))
+    .map((questPair) => questPair[0]);
+}
+
+export function startQuest(questPref: string) {
+  const kioskText = visitUrl(kioskUrl);
+  const questChoice = Array.from(questPreferenceKioskText)
+    .filter((questPair) => kioskText.indexOf(questPair[1]) > -1)
+    .sort(
+      (questPairA, questPairB) =>
+        kioskText.indexOf(questPairA[1]) - kioskText.indexOf(questPairB[1])
+    )
+    .map((questPair) => questPair[0])
+    .indexOf(questPref);
+  if (questChoice > -1) {
+    runChoice(questChoice + 1);
+  }
+}
+
+export function finishQuest() {
+  const currentQuest = acceptedQuest();
+  if (currentQuest) {
+    const currentQuestRequirement = questRequirements.get(currentQuest)!;
+    if (
+      currentQuestRequirement.trackingPref === "" ||
+      getNumber(currentQuestRequirement.trackingPref) ===
+        currentQuestRequirement.completeValue
+    ) {
+      visitUrl(kioskUrl);
+      if (3 in availableChoiceOptions()) {
+        runChoice(3);
+      }
+    }
+  }
+}
+
+export function coasterNextTurn() {
+  return get("dinseyRollercoasterNext");
+}
+
+export function canFightDinsey() {
+  return (
+    Array.from(keyCardsLocations.keys()).every((keycard) => have(keycard)) &&
+    get("lastWartDinseyDefeated") !== myAscensions() &&
+    getRemainingLiver() >= 0 &&
+    myAdventures() > 0
+  );
+}
+
+export function disposeGarbage() {
+  if (have($item`bag of park garbage`) && !get("_dinseyGarbageDisposed")) {
+    visitUrl(maintUrl);
+    runChoice(6);
+  }
+}
+
+export function fightDinsey() {
```

Comment:
I would probably add a macro input to this so you don't have to wrap it with a withMacro

Reply:
I did not do this because how to submit combat is not defined in Libram, so I let the user choose how to submit combat

Reply:
One final remark: I would probably call this `fightWart` on the basis that `Dinsey.fightDinsey()` feels redundant

Reply:
oh you renamed it
ignore me!